core/common/src/main/java/alluxio/security/group/GroupMapping.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.security.group;
+
+import alluxio.Configuration;
+import alluxio.Constants;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Class to map user to groups. It maintain a cache for user to groups mapping. The underlying
+ * implementation depends on {@link GroupMappingService}.
+ */
+public class GroupMapping {
```

Comment:
Yeah, I've renamed to `CachedGroupMapping`, and re-organized the structure.

`GroupMappingService` is still the interface that developers can implement.
Its factory can provide `getCachedGroupMapping`